modules/kernel/src/test/scala/org/tesselation/kernel/CellMonoidLawsSuite.scala

```diff
@@ -0,0 +1,34 @@
+package org.tesselation.kernel
+
+import cats.Monoid
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import cats.kernel.Eq
+import cats.kernel.laws.discipline.MonoidTests
+
+import org.tesselation.kernel.CellMonoidManualTestSuite.TestCells.{LongToStringCell, TestLongInput}
+
+import org.scalacheck.{Arbitrary, Gen}
+import weaver.FunSuite
+import weaver.discipline.Discipline
+
+object CellMonoidLawsSuite extends FunSuite with Discipline {
+  implicit val M: Monoid[Cell[IO, StackF, Ω, Either[CellError, Ω], Ω]] = Cell.cellMonoid[IO, StackF]
+
+  // WARN: By comparing cell's run method result we can consider monoid for cells fulfilling the monoid laws.
+  //       Because two functions with the same implementation are never equal to each other we can't compare
+  //       the cells to each other by comparing the cell objects (neither convert nor hylo functions will be ever equal).
+  implicit val eq: Eq[Cell[IO, StackF, Ω, Either[CellError, Ω], Ω]] =
+    (x: Cell[IO, StackF, Ω, Either[CellError, Ω], Ω], y: Cell[IO, StackF, Ω, Either[CellError, Ω], Ω]) => {
+      for {
+        xResult <- x.run()
+        yResult <- y.run()
+      } yield xResult == yResult
+    }.unsafeRunSync()
```

Comment:
IMO we should write it without this unsafe run sync

Reply:
Maybe I could test it with a different monad 🤔

Reply:
Switched to Id, will push the update in a second.